pkg/k8s/events/crud.go

```diff
@@ -37,23 +37,41 @@ func List(ctx context.Context, namespace, podName string, c kubernetes.Interface
 	return events.Items, err
 }
 
-func GetUnhealthyEventFailure(ctx context.Context, namespace, podName string, c kubernetes.Interface) string {
+const (
+	unhealthyReason             = "Unhealthy"
+	readinessProbeFailedMessage = "Readiness probe failed:"
+	livenessProbeFailedMessage  = "Liveness probe failed:"
+)
+
+func readinessProbeFailed(event apiv1.Event) bool {
+	return event.Reason == unhealthyReason && strings.HasPrefix(event.Message, readinessProbeFailedMessage)
+}
+func livenessProbeFailed(event apiv1.Event) bool {
+	return event.Reason == unhealthyReason && strings.HasPrefix(event.Message, livenessProbeFailedMessage)
+}
+
+type HealthcheckFailure struct {
+	Readiness string
+	Liveness  string
+}
+
+// GetUnhealthyEventFailure returns the message of the last event that caused the pod to be unhealthy
+// this could be a readiness or liveness probe failure
+func GetUnhealthyEventFailure(ctx context.Context, namespace, podName string, c kubernetes.Interface) *HealthcheckFailure {
 	events, err := List(ctx, namespace, podName, c)
 	if err != nil {
-		return ""
+		return nil
 	}
-	previousKilling := false
 	for i := len(events) - 1; i >= 0; i-- {
-		if previousKilling {
-			if events[i].Reason == "Unhealthy" && strings.Contains(events[i].Message, "probe failed") {
-				return events[i].Message
-			}
+		event := events[i]
```

Comment:
Why do we remove the check about the killing pod event?

Reply:
the event we want to check is the liveness probe being failed. After looking into the logic, i dont see how checking if "previous" was a killing event adds value to the output of the probe having failed ?

Reply:
updated the code to include this condition. i assumed the killing event was not need as a condition to check the events what that the restart counts is greater than 0, but these do not reflect exactly that it has been killed.